### DISCUSSION TITLE
Migrating a self-hosted Bitwarden to Vaultwarden

### DISCUSSION THREAD

**Discussion by ferdzo** [dani-garcia/vaultwarden#5302] [Q&A]
Hello,

I currently have a self-hosted Bitwarden with organizations, and I want to migrate my whole Bitwarden to a new self-hosted Vaultwarden. But can't find any resources for this. First of all, is it even possible to migrate it? Thanks.

**Comment by BlackDex** [comment]
Export and import using Bitwarden JSON format.
This needs to be done per org and per personal vault.

All users and orgs need to be registered and invited and configured manually again

**Reply by stefan0xC** [reply]
Also any attachments have to be taken care of manually as they are not exported.
Cf. https://bitwarden.com/help/export-your-data/

**Reply by BlackDex** [reply]
Good one

**Comment by ghaisasadvait** [comment]
Any way to import everything automatically from the ms sql database and supplying DB password etc?

**Reply by stefan0xC** [reply]
Nope, at least nothing I am aware of. And I can't imagine why there would be something like that as both backends are not using the same database layout.

**Reply by BlackDex** [reply]
No not really. The two databases are not really compatible. 
And Bitwarden uses some database stored procedures and other features to adjust the database data server side.

**Reply by ghaisasadvait** [reply]
Noted. I was really hoping there was a way to migrate it as I cannot ask each user to setup account agin from scratch. Thanks for the clarifications!
